subsys/sensing/sensor_mgmt.h

```diff
@@ -154,6 +100,40 @@ static inline const struct sensing_sensor_info *get_sensor_info(struct sensing_c
 	return conn->source->info;
 }
 
+/* check if client has requested data from reporter */
+static inline bool is_client_request_data(struct sensing_connection *conn)
+{
+	return conn->interval != 0;
+}
+
+static inline uint64_t get_us(void)
+{
+	return k_ticks_to_us_floor64(k_uptime_ticks());
```

Comment:
Ticks are likely going to represent in the most common case millisecond levels of precision. So I'm skeptical that this get_us(void) function will be able to do what we need it to.

Likely what we want is k_cycle_get_64() and converting the cycle count to microseconds. Or perhaps doing some math to get the cycles since the last tick (maybe we need this functionality) so that sub-tick timing precision without rollovers can be obtained. This would be useful for sensor drivers as well I believe as they need a way to precisely timestamp the data.

Reply:
Agreed, I'm OK with timing for now just moving forward with more of a "whatever" until https://github.com/zephyrproject-rtos/zephyr/issues/63651 is resolved. I think we're likely to land on cycles like @teburd  said.